### PR TITLE
fix uvbeam.interp issue.

### LINKED ISSUE #16: [BUG] incorrect `uvbeam.interp` call in `conversions.py`
As we can see from

https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/4c66208245db29be8b9920434d6d48d23b29e367/pyuvdata/uvbeam/uvbeam.py#L1579

`uvbeam.interp` takes `az_array` and `za_array` as its first two args but `freq_array` is now the sixth arg. 

Unfortunately, `conversions.py` calls `uvbeam.interp` assuming that `freq_array=freqs` is the third arg

https://github.com/HERA-Team/hera_sim/blob/144f8ec8033e45ace1a30e3f1f45c8a2489512b3/hera_sim/visibilities/conversions.py#L36

which assigns `freqs` to `az_za_grid` in `uvbeam.interp` and causes all kinds nonsense.

This line should be changed to name the args.

` efield_beam = uvbeam.interp(az_array=az, za_array=za, freq_array=freqs, **kwargs)[0]`

### LINKED ISSUE #18: [Bug / Feature Request] negative azimuth values returned by `enu_to_az_za` raise interpolation errors
The standard function from converting `enu` to `az_za` in `vis_cpu` 

https://github.com/HERA-Team/vis_cpu/blob/b602819508dee05742014b99dfce363a57cc4ffd/src/vis_cpu/conversions.py#L10

which is used to produce azimuths to be interpolated by `uvbeam`

https://github.com/HERA-Team/vis_cpu/blob/b602819508dee05742014b99dfce363a57cc4ffd/src/vis_cpu/conversions.py#L286

outputs azimuths between -pi and pi.

Many `uvbeam` objects can be expected to only have positive azimuths (for example, cst beams only run from 0 to 2pi) and this raises the error in `uvbeam` at

https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/4c66208245db29be8b9920434d6d48d23b29e367/pyuvdata/uvbeam/uvbeam.py#L1387

This could be avoided if `uvbeam` recognized negative azimuth values as being identical to positive azimuth values (mod 2 pi) or if we constrained the azimuth to bet between 0 and 2 pi in `vis_cpu.conversions.enu_to_az_za`

### PR DESCRIPTION
**Pull request by aewallwi**
closes #16 
closes #18

### CODE REVIEW

File: src/vis_cpu/conversions.py [hunk 1]
```diff
@@ -52,7 +59,8 @@ def enu_to_az_za(enu_e, enu_n, orientation="astropy"):
     # Flip and rotate azimuth coordinate if uvbeam convention is used
     if orientation == "uvbeam":
         az = 0.5 * np.pi - az
-
+    if constraint_az_between_zero_twopi:
```
Comment (philbull) [inline src/vis_cpu/conversions.py:62]: Is there any reason why we shouldn't just leave this on by default? I don't see any big issues with forcing Azimuth to be in this range.

**Review by philbull — CHANGES_REQUESTED** [review]
Thanks for fixing this -- just a few minor comments.

File: src/vis_cpu/conversions.py [hunk 2]
```diff
@@ -7,7 +7,9 @@
 from astropy.time import Time
 
 
-def enu_to_az_za(enu_e, enu_n, orientation="astropy"):
+def enu_to_az_za(
+    enu_e, enu_n, orientation="astropy", constraint_az_between_zero_twopi=False
```
[the remark this answers is not in the document]
Comment (aewallwi) [reply src/vis_cpu/conversions.py:11]: I think this makes sense.

File: src/vis_cpu/conversions.py [hunk 3]
```diff
@@ -33,6 +35,11 @@ def enu_to_az_za(enu_e, enu_n, orientation="astropy"):
         Az(E) = +90 deg). Alternatively, the ``'uvbeam'`` convention uses
         North of East (Az(N) = +90 deg, Az(E) = 0 deg).
 
+    constraint_az_between_zero_twopi : bool, optional
+        if True, constrain az to be betwee 0 and 2 * pi
+        This avoids negative az values which are standard for outputs of arctan2
+        and avoids problems that can arise with
```
[the remark this answers is not in the document]
Comment (aewallwi) [reply src/vis_cpu/conversions.py:41]: completed.

File: src/vis_cpu/conversions.py (hunk 1 above)
Comment (aewallwi) [reply src/vis_cpu/conversions.py:62]: I agree with making the default range 0-2pi (its much more sensible). Off the bat, I didn't want to change the default behavior without consultation but if others are for it I think we should.

**Comment by aewallwi** [comment]
Thanks for the review @philbull. This should be ready for another look whenever you have a chance.

**Review by philbull — APPROVED** [review]

**Pull request merged**